### PR TITLE
✨ #613 all apps menu dropdown

### PR DESCRIPTION
**Pull request by ciaranschutte**

### CODE REVIEW

File: src/components/Header/AppsMenu.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  DropdownExternalLink,
+  DropdownRow,
+  AllAppsLabelContainer,
+  MenuLabelContainer,
+} from './ui';
+import AllAppsContactIcon from '../../icons/AllAppsContactIcon';
+import AllAppsSupportIcon from '../../icons/AllAppsSupportIcon';
+import AllAppsWebsiteIcon from '../../icons/AllAppsWebsiteIcon';
+import AllAppsPortalIcon from '../../icons/AllAppsPortalIcon';
+import AllAppsMenuIcon from '../../icons/AllAppsMenuIcon';
```
Comment (hlminh2000) [inline src/components/Header/AppsMenu.js]: can probably be referenced as `icons/...`

File: src/components/Header/AppsMenu.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+);
+
+export default compose(
+  withState('isDropdownVisible', 'setDropdownVisibility', false),
+  withHandlers({
+    toggleDropdown: ({ isDropdownVisible, setDropdownVisibility }) => e => {
+      setDropdownVisibility(!isDropdownVisible);
+    },
+  }),
+)(AppsMenu);
```
Comment (hlminh2000) [inline src/components/Header/AppsMenu.js]: I prefer render props over hoc personally, I find it actually reduces the cost of moving state around much easier... In this case, I'd just use `react-component-component` for these kinds of state

Also for your entertainment, check out this article Chang wrote a while back about the namespacing problem with hoc :
https://hackernoon.com/solving-the-problems-of-higher-order-components-without-throwing-the-baby-out-with-the-bathwater-40ddc72df5aa

File: src/components/Header/ui.js [hunk 3]
```diff
@@ -4,9 +4,12 @@ import Row from 'uikit/Row';
 import Gravtar from 'uikit/Gravatar';
 import { DropdownContainer, DropdownOptionsContainer } from 'uikit/Dropdown';
 import { space } from 'styled-system';
+import { DropdownLabelContainer } from '../../uikit/Dropdown/ui';
```
Comment (hlminh2000) [inline src/components/Header/ui.js]: `uikit/Dropdown/ui`? :)

File: src/components/Header/AppsMenu.js (hunk 2 above)
Comment (ciaranschutte) [reply src/components/Header/AppsMenu.js]: ```const hocToProvider = hoc => hoc(({ render, ...props }) => render(props));```

**Pull request merged**